tests/run-make/core-ffi-typecheck-clang/rmake.rs

```diff
@@ -0,0 +1,235 @@
+            const CLANG_C_SIZE_T_SIZE: usize = {};
+            const CLANG_C_PTRDIFF_T_SIZE: usize = {};
+            ",
+            parse_size(&defines, "CHAR"),
+            char_is_signed(&defines),
+            parse_size(&defines, "SHORT"),
+            parse_size(&defines, "INT"),
+            parse_size(&defines, "LONG"),
+            parse_size(&defines, "LONG_LONG"),
+            parse_size(&defines, "FLOAT"),
+            parse_size(&defines, "DOUBLE"),
+            parse_size(&defines, "SIZE_T"),
+            parse_size(&defines, "PTRDIFF_T"),
+        ));
+
+        // Generate a target-specific rmake file.
+        // If type misalignments occur,
+        // generated rmake file name used to identify the failing target.
+        let file_name = format!("{}_rmake.rs", target.replace("-", "_").replace(".", "_"));
+
+        // Attempt to build the test file for the relevant target.
+        // Tests use constant evaluation, so running is not necessary.
+        rfs::write(&file_name, rmake_content);
+        let rustc_output = rustc()
+            .arg("-Zunstable-options")
+            .arg("--emit=metadata")
+            .arg("--target")
+            .arg(target)
+            .arg("-o-")
+            .arg(&file_name)
+            .run();
+        rfs::remove_file(&file_name);
+        if !rustc_output.status().success() {
+            panic!("Failed for target {}", target);
+        }
+    }
+}
+
+// Helper to parse size from clang defines
+fn parse_size(defines: &str, type_name: &str) -> usize {
+    let search_pattern = format!("__SIZEOF_{}__ ", type_name.to_uppercase());
+    for line in defines.lines() {
+        if line.contains(&search_pattern) {
+            if let Some(size_str) = line.split_whitespace().last() {
+                return size_str.parse().unwrap_or(0);
+            }
+        }
+    }
+
+    // Only allow CHAR to default to 1
+    if type_name.to_uppercase() == "CHAR" {
+        return 1;
+    }
+
+    panic!("Could not find size definition for type: {}", type_name);
+}
+
+fn char_is_signed(defines: &str) -> bool {
+    !defines.lines().any(|line| line.contains("__CHAR_UNSIGNED__"))
+}
+
+/// Parse core/ffi/mod.rs to retrieve only necessary macros and type defines
+fn preprocess_core_ffi() {
+    let mod_path = run_make_support::source_root().join("library/core/src/ffi/primitives.rs");
+    let mut content = rfs::read_to_string(&mod_path);
+
+    //remove stability features #![unstable]
+    let mut re = regex::Regex::new(r"#!?\[(un)?stable[^]]*?\]").unwrap();
+    content = re.replace_all(&content, "").to_string();
+
+    //remove doc features #[doc...]
+    re = regex::Regex::new(r"#\[doc[^]]*?\]").unwrap();
+    content = re.replace_all(&content, "").to_string();
+
+    //remove non inline modules
+    re = regex::Regex::new(r".*mod.*;").unwrap();
+    content = re.replace_all(&content, "").to_string();
+
+    let file_name = "processed_mod.rs";
+
+    rfs::write(&file_name, content);
```

Comment:
Suggestion: write a line of `#![allow(non_camel_case_types)]` before the stripped `primitives.rs` content to trim down amount of warnings if the test do fail.